app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/detail/DetailScreenUiTest.kt

```diff
@@ -0,0 +1,155 @@
+package co.nimblehq.compose.crypto.ui.screen.detail
+
+import androidx.activity.compose.setContent
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinDetailUseCase
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinPricesUseCase
+import co.nimblehq.compose.crypto.extension.toFormattedString
+import co.nimblehq.compose.crypto.test.MockUtil.coinDetail
+import co.nimblehq.compose.crypto.test.MockUtil.coinPrices
+import co.nimblehq.compose.crypto.ui.BaseScreenTest
+import co.nimblehq.compose.crypto.ui.components.chartintervals.TimeIntervals
+import co.nimblehq.compose.crypto.ui.navigation.AppDestination
+import co.nimblehq.compose.crypto.ui.screens.MainActivity
+import co.nimblehq.compose.crypto.ui.screens.detail.*
+import co.nimblehq.compose.crypto.ui.screens.home.FIAT_CURRENCY
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.math.abs
+
+@ExperimentalCoroutinesApi
+class DetailScreenUiTest : BaseScreenTest() {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetCoinDetailUseCase: GetCoinDetailUseCase = mockk()
+    private val mockGetCoinPricesUseCase: GetCoinPricesUseCase = mockk()
+
+    private lateinit var detailViewModel: DetailViewModel
+
+    private var appDestination: AppDestination? = null
+
+    @Before
+    fun setUp() {
+
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
+
+        initViewModel()
+        composeAndroidTestRule.activity.setContent {
+            DetailScreen(
+                coinId = "",
+                viewModel = detailViewModel,
+                navigator = { destination ->
+                    appDestination = destination
+                }
```

Comment:
Could you please check if we could get rid of the `appDestination`? 🤔 

```suggestion
                navigator = {}
```